internal/workflow.go

```diff
@@ -297,6 +297,17 @@ type (
 		fn       interface{}
 		isMethod bool
 	}
+
+	// UpdateOptions consists of options for executing a named workflow update.
+	UpdateOptions struct {
+		// Validator is an optional (i.e. can be left nil) func with exactly the
+		// same type signature as the required update handler func but returning
+		// only a single value of type error. The implementation of this
+		// function MUST be safe and MUST NOT alter workflow state in any way.
+		// A panic from this function will be treated as equivalent to returning
+		// an error.
```

Comment:
I'm going to remove "safe" since it's correct but jargonny and "MUST NOT alter workflow state" is the same thing.

Do users think in terms of "generating commands?" It would be good to provide an exhaustive list of prohibited functions here but it would get out of date quickly. I'll give some examples anyhow.